java/client/src/main/java/glide/api/models/commands/WeightAggregateOptions.java

```diff
@@ -43,26 +31,64 @@ public enum Aggregate {
         MIN,
         /** Aggregates by selecting the maximum score for each element across sets. */
         MAX;
+
+        public String[] toArgs() {
+            return new String[] {AGGREGATE_REDIS_API, toString()};
+        }
     }
 
     /**
-     * Converts WeightAggregateOptions into a String[].
+     * Basic interface. Please use one of the following implementations:
      *
-     * @return String[]
+     * <ul>
+     *   <li>{@link KeyArray}
+     *   <li>{@link WeightedKeys}
+     * </ul>
      */
-    public String[] toArgs() {
-        List<String> optionArgs = new ArrayList<>();
+    public interface KeysOrWeightedKeys {
+        String[] toArgs();
+    }
 
-        if (!weights.isEmpty()) {
-            optionArgs.add(WEIGHTS_REDIS_API);
-            optionArgs.addAll(
-                    weights.stream().map(element -> Double.toString(element)).collect(Collectors.toList()));
-        }
+    /** Represents the keys of the sorted sets involved in the aggregation operation. */
+    @RequiredArgsConstructor
+    public static class KeyArray implements KeysOrWeightedKeys {
+        private final String[] keys;
 
-        if (aggregate != null) {
-            optionArgs.addAll(List.of(AGGREGATE_REDIS_API, aggregate.toString()));
+        /** Convert to command arguments according to the Redis API. */
```

Comment:
If you move this doc to the interface, you don't need to repeate it for every implementation